services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -407,10 +411,13 @@ public Integer getUsersCount(@QueryParam("search") String search,
             if (enabled != null) {
                 parameters.put(UserModel.ENABLED, enabled.toString());
             }
+            parameters.putAll(searchAttributes);
+            Stream<UserModel> users = session.users().searchForUserStream(realm, parameters, -1, -1);
```

Comment:
There are `getUsersCount()`-methods to get count of users in the UserQueryProvider. Please use them instead of replicating them here.

Reply:
Hey @sschu 
I have added test for this. 
But when I use from `session.users().getUsersCount(...)` I get wrong result (always return total number of users).

Because of this I have used `session.users().searchForUserStream(...)`

Could you take a look?

Reply:
@gilvansfilho thanks for this patch it solves a problem for us which is to paginate based on searching by users attributes and search fields at the same time.

I tried to see if I could put the code back to use session.users().getUsersCount(realm, params) but the problem is that there are multiple implementations of UserQueryProvider.

I think the best way is to look at how the UsersResource.getUsers call was enhanced to allow both the q parameter and other parameters to be used at the same time.

[5b0986e490090433658e9a1e607c0e248f4cc756](https://github.com/keycloak/keycloak/commit/5b0986e490090433658e9a1e607c0e248f4cc756)

You can see that the JpaUserProvider and MapUserProvider where enhanced so that they process the q search parameter.

So the problem within using the existing methods is that both of these providers need to be enhanced in a similar way to process the q search parameter for the getUsersCount(realm, params) method.